### PR TITLE
Use sbt 1 to build Play itself

### LINKED ISSUE #8383: Use sbt 1 to build Play itself
### Play Version

Play 2.6+.

Maybe this won't be backported to Play 2.6, but there aren't huge differences between Play 2.6 and master branch regarding the build configuration, so we should be okay to backport.

### Expected Behavior

Play already cross build its plugin to sbt 0.13 and 1, but Play itself is not build using sbt 1. sbt tracks the progress of projects already using sbt 1 here:

https://github.com/sbt/sbt/wiki/library-sbt-1.x-migration

Not to be confused with [plugins migration](https://github.com/sbt/sbt/wiki/sbt-1.x-plugin-migration).

### Actual Behavior

Play is not built with sbt 1.

### Related

1. https://github.com/playframework/interplay/pull/43

### PR DESCRIPTION
**Pull request by gaurang-sawhney**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [ ] Have you added copyright headers to new files?
* [ ] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [ ] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #8383 

## Purpose

Updated the sbt version to `1.2.8`.

## Background Context

[Link to discussion](https://github.com/playframework/playframework/issues/8383#issuecomment-458431728)

## References

Are there any relevant issues / PRs / mailing lists discussions?

### CODE REVIEW

File: framework/project/Dependencies.scala [hunk 1]
```diff
@@ -229,7 +231,8 @@ object Dependencies {
 
   val playDocVersion = "2.0.0"
   val playDocsDependencies = Seq(
-    "com.typesafe.play" %% "play-doc" % playDocVersion
+    "com.typesafe.play" %% "play-doc" % playDocVersion,
+    "org.scala-sbt" % "compiler-bridge_2.12" % Dependencies.compilerBridgeVersion % Configurations.ScalaTool
```
[the remark this answers is not in the document]
Comment (gaurang-sawhney) [reply framework/project/Dependencies.scala]: While generating the docs using `Docs.scala`, we try to get the instance of Analyzing compiler where it internally calls out `xsbt.ScaladocInterface` which resides in `compiler-bridge.`
Ideally `compiler-bridge` should be fetched using `scalaCompilerBridgeBinaryJar` task provided by `sbt`, however this is returning `None`. So just for experiment, I tried to add `compiler-bridge` explicitly but it doesn't help either.

File: framework/project/Docs.scala [hunk 2]
```diff
@@ -24,6 +33,15 @@ object Docs {
   val allConfs = TaskKey[Seq[(String, File)]]("allConfs", "Gather all configuration files")
 
   lazy val settings = Seq(
+    // Temporary override since sbt is unable to update with dynamically added dependency
+    scalaCompilerBridgeBinaryJar := {
```
[the remark this answers is not in the document]
Comment (gaurang-sawhney) [reply framework/project/Docs.scala]: As I mentioned above, this is was an effort to fetch the `compiler-bridge` by overriding the `scalaCompilerBridgeBinaryJar` task provided by `sbt` but again this doesn't help either.

File: framework/project/Docs.scala [hunk 3]
```diff
@@ -134,15 +151,22 @@ object Docs {
       val compilers = Keys.compilers.value
       val useCache = apiDocsUseCache.value
       val classpath = apiDocsClasspath.value
+      val apiDocsScalaSources = Docs.apiDocsScalaSources.value
+      val apiDocsJavaSources = Docs.apiDocsJavaSources.value
+      val streams = sbt.Keys.streams.value
+
+      val scalaCompiler = compilers.scalac()
+      val provider = ZincCompilerUtil.constantBridgeProvider(scalaCompiler.scalaInstance, new File("random"))
+      val scalac = new AnalyzingCompiler(scalaCompiler.scalaInstance, provider, scalaCompiler.classpathOptions(), (a: Seq[String]) => (), None)
```
[the remark this answers is not in the document]
Comment (gaurang-sawhney) [reply framework/project/Docs.scala]: With `sbt 0.13`, `compilers.scalac()` used to return `AnalyzingCompiler`, but with `sbt 1.0`, it now returns `ScalaCompiler` and there is no one to one mapping of `ScalaCompiler` and `AnalyzingCompiler`.

File: framework/project/BuildSettings.scala [hunk 4]
```diff
@@ -106,7 +111,8 @@ object BuildSettings {
         Resolver.typesafeRepo("releases"),
         Resolver.typesafeIvyRepo("releases")
       ),
-      javacOptions ++= Seq("-encoding",  "UTF-8", "-Xlint:unchecked", "-Xlint:deprecation"),
+      ivyConfigurations ++= Seq(DocsApplication, SourcesApplication),
+    javacOptions ++= Seq("-encoding",  "UTF-8", "-Xlint:unchecked", "-Xlint:deprecation"),
```
[the remark this answers is not in the document]
Comment (gaurang-sawhney) [reply framework/project/BuildSettings.scala]: 👍

File: framework/project/Tasks.scala [hunk 5]
```diff
@@ -60,11 +60,11 @@ object Commands {
       state.log.info("Turning on quick publish")
     }
 
-    val newStructure = Load.reapply(filtered ++ Seq(
+    x.appendWithoutSession(filtered ++ Seq(
```
[the remark this answers is not in the document]
Comment (gaurang-sawhney) [reply framework/project/Tasks.scala]: 👍

**Comment by gaurang-sawhney** [comment]
@marcospereira The binary compatibility check is failing here due to unavailable binary for scala version `2.13.0-M5` of `compiler-bridge:1.2.5` which is the latest release.

`compiler-bridge:1.2.5` supports `2.13.0-M2`.
`compiler-bridge:1.3.0-M2` supports `2.13.0-M5` but I am not sure if we should depend on milestone releases.

**Comment by gaurang-sawhney** [comment]
@dwijnand Any guidance about [the above comment](https://github.com/playframework/playframework/pull/9001#issuecomment-462613969)

**Comment by dwijnand** [comment]
Hey @gaurang-sawhney, I hope to look into this in the next weeks.

**Comment by marcospereira** [comment]
@dwijnand could you please review this?

**Comment by dwijnand** [comment]
2.7.x 2729a9c49
2.6.x https://github.com/playframework/playframework/pull/9140

**Pull request merged**